airflow/serialization/serialized_objects.py

```diff
@@ -395,6 +411,59 @@ def deserialize_operator(cls, encoded_op: Dict[str, Any]) -> BaseOperator:
 
         return op
 
+    @classmethod
+    def _is_constcutor_param(cls, attrname: str, instance: Any) -> bool:
+        # Check all super classes too
+        return any(
+            attrname in cls.__constructor_params_for_subclass(typ)
+            for typ in type(instance).mro()
+        )
+
+    @classmethod
+    def _value_is_hardcoded_default(cls, attrname: str, value: Any, instance: Any) -> bool:
+        """
+        Check if ``value`` is the default value for ``attrname`` as set by the
+        constructor of ``instance``, or any of it's parent classes up
+        to-and-including BaseOperator.
+
+        .. seealso::
+
+            :py:meth:`BaseSerialization._value_is_hardcoded_default`
+        """
+
+        def _is_default():
+            nonlocal ctor_params, attrname, value
```

Comment:
This won't work on python2, instead we'd have to do

```
        def _is_default(ctor_params, attrname, value):
```

and pass em in at the call site

Reply:
I went through a number of iterations of way of doing this, it may be clearer just to pass them in anyway on Py3. WDYT?

Reply:
Yeah let's just pass them in

Reply:
Done